lib/evm_database_ops.rb

```diff
@@ -116,13 +118,15 @@ def self.restore(db_opts, connect_opts = {})
       end
 
       session = MiqGenericMountSession.new_session(connect_opts)
-      db_opts[:local_file] = session.uri_to_local_path(uri)
+      db_opts[:local_file]  = session.uri_to_local_path(uri)
+      db_opts[:object_file] = session.uri_to_object_path(uri) if connect_opts[:object_store]
+      db_opts[:session]     = session if db_opts[:session].class.name == "MiqS3Session"
     end
 
     block_result = yield db_opts if block_given?
     uri || block_result
   ensure
-    session.disconnect if session
+    session&.disconnect
```

Comment:
Why not just leave this as is?

Reply:
Can do.  Previously there was another line that we moved that called the add method as well.